addons/website_sale_delivery/models/sale_order.py

```diff
@@ -124,66 +86,60 @@ def _get_delivery_methods(self, cr, uid, order, context=None):
                 # RIM: hack to remove in master, because available field should not depend on a SOAP call to external shipping provider
                 # The validation error is used in backend to display errors in fedex config, but should fail silently in frontend
                 _logger.debug("%s removed from e-commerce carrier list. %s" % (carrier.name, e))
-
         return available_carrier_ids
 
-    def _get_errors(self, cr, uid, order, context=None):
-        errors = super(SaleOrder, self)._get_errors(cr, uid, order, context=context)
-        if not self._get_delivery_methods(cr, uid, order, context=context):
+    @api.model
+    def _get_errors(self, order):
+        errors = super(SaleOrder, self)._get_errors(order)
+        if not order._get_delivery_methods():
             errors.append(
                 (_('Sorry, we are unable to ship your order'),
                  _('No shipping method is available for your current order and shipping address. '
                    'Please contact us for more information.')))
         return errors
 
-    def _get_website_data(self, cr, uid, order, context=None):
+    @api.model
+    def _get_website_data(self, order):
         """ Override to add delivery-related website data. """
-        values = super(SaleOrder, self)._get_website_data(cr, uid, order, context=context)
+        values = super(SaleOrder, self)._get_website_data(order)
         # We need a delivery only if we have stockable products
-        has_stockable_products = False
-        for line in order.order_line:
-            if line.product_id.type in ('consu', 'product'):
-                has_stockable_products = True
-        if not has_stockable_products:
+        if not order.order_line.filtered(lambda line: line.product_id.type in ['consu', 'product']):
             return values
 
-        delivery_ctx = dict(context, order_id=order.id)
-        DeliveryCarrier = self.pool.get('delivery.carrier')
-        delivery_ids = self._get_delivery_methods(cr, uid, order, context=context)
+        delivery_ids = order._get_delivery_methods()
 
-        values['deliveries'] = DeliveryCarrier.browse(cr, SUPERUSER_ID, delivery_ids, context=delivery_ctx)
+        values['deliveries'] = self.env['delivery.carrier'].sudo().with_context(order_id=order.id).browse(delivery_ids)
         return values
 
-    def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, set_qty=0, context=None, **kwargs):
+    @api.multi
+    def _cart_update(self, product_id=None, line_id=None, add_qty=0, set_qty=0, **kwargs):
```

Comment:
Yes, the super method itself is in old API context and called from website_sale, website_event_sale and website_sale_options.